File: sdk/kotlin/src/main/kotlin/com/neo/sdk/NeoClient.kt
package com.neo.sdk

class NeoClient(
    private val host: String = "localhost",
    private val port: Int = 8080,
    private val apiKey: String? = null,
) {
    private var connected: Boolean = false
    val isConnected: Boolean get() = connected

    fun connect() {
        connected = true
    }

    fun disconnect() {
        connected = false
    }

    fun createAgent(name: String, config: Map<String, Any> = emptyMap()): AgentHandle {
        require(connected) { "Client not connected" }
        return AgentHandle(
            id = java.util.UUID.randomUUID().toString(),
            name = name,
            state = "idle",
            config = config,
        )
    }

    fun submitTask(agentId: String, task: Map<String, Any>): TaskHandle {
        require(connected) { "Client not connected" }
        return TaskHandle(
            id = java.util.UUID.randomUUID().toString(),
            agentId = agentId,
            status = "pending",
            payload = task,
        )
    }

    fun health(): Map<String, Any> = mapOf(
        "status" to if (connected) "ok" else "disconnected",
        "host" to host,
        "port" to port,
    )
}

data class AgentHandle(
    val id: String,
    val name: String,
    val state: String,
    val config: Map<String, Any> = emptyMap(),
)

data class TaskHandle(
    val id: String,
    val agentId: String,
    val status: String,
    val payload: Map<String, Any> = emptyMap(),
    val result: Any? = null,
)

File: sdk/swift/Sources/NeoSDK/NeoClient.swift
import Foundation

public class NeoClient {
    public struct AgentHandle {
        public let id: String
        public let name: String
        public let state: String
    }

    public struct TaskHandle {
        public let id: String
        public let agentId: String
        public let status: String
    }

    private let host: String
    private let port: Int
    private let apiKey: String?
    public private(set) var isConnected: Bool = false

    public init(host: String = "localhost", port: Int = 8080, apiKey: String? = nil) {
        self.host = host
        self.port = port
        self.apiKey = apiKey
    }

    public func connect() {
        isConnected = true
    }

    public func disconnect() {
        isConnected = false
    }

    public func createAgent(name: String) -> AgentHandle {
        precondition(isConnected, "Client not connected")
        return AgentHandle(
            id: UUID().uuidString,
            name: name,
            state: "idle"
        )
    }

    public func submitTask(agentId: String, payload: [String: Any]) -> TaskHandle {
        precondition(isConnected, "Client not connected")
        return TaskHandle(
            id: UUID().uuidString,
            agentId: agentId,
            status: "pending"
        )
    }

    public func health() -> [String: Any] {
        return [
            "status": isConnected ? "ok" : "disconnected",
            "host": host,
            "port": port,
        ]
    }
}
